### PR TITLE
move the code into the main function 

### LINKED ISSUE #21: Move code into a main function
Currently the main code for the game is defined at the module level, outside of any function. I.e., the start of the module looks like this at the moment...

```
import random
import getpass
import pycountry
from countryinfo import CountryInfo

# Fetch all countries and their capitals using pycountry
countries_capitals = {}
for country in pycountry.countries:
    try:
        country_info = CountryInfo(country.alpha_2)
        capital = country_info.capital()
        if capital:
            countries_capitals[country.name] = capital
    except (KeyError, ValueError):
        continue


# Function to choose a random country from the dictionary
def choose_country():
    country = random.choice(list(countries_capitals.keys()))
    return country
    

# Function to check if the guess is correct
def check_guess(guess, country):
    capital = countries_capitals[country]
    if guess.lower() == capital.lower():
        return True
    else:
        return False

# Initialize player scores
player1_score = 0
player2_score = 0

#Defining the congratulatory message
congrats_message ="Congratulations Player {winner}, you are the winner!"

[... rest of the main game code ...]
```

Putting code at the module level like this, outside of a function, can cause some problems. For example, currently flake8 is not picking up the fact that the `congrats_message` variable is declared but never used (accessed).

It would be better to move the main logic into a function. Something like this would be conventional...

```
import random
import getpass
import pycountry
from countryinfo import CountryInfo

# Function to choose a random country from the dictionary
def choose_country():
    country = random.choice(list(countries_capitals.keys()))
    return country
    

# Function to check if the guess is correct
def check_guess(guess, country):
    capital = countries_capitals[country]
    if guess.lower() == capital.lower():
        return True
    else:
        return False


def main():
    """Main game logic."""

    # Fetch all countries and their capitals using pycountry
    countries_capitals = {}
    for country in pycountry.countries:
        try:
            country_info = CountryInfo(country.alpha_2)
            capital = country_info.capital()
            if capital:
                countries_capitals[country.name] = capital
        except (KeyError, ValueError):
            continue

    # Initialize player scores
    player1_score = 0
    player2_score = 0
    
    #Defining the congratulatory message
    congrats_message ="Congratulations Player {winner}, you are the winner!"

    [... rest of the main game code ...]


if __name__ == "__main__":
    # Call the main function if script has been run from the command line.
    main()
```

I.e., here all of the main game logic is moved inside a new function called `main()`. 

Then, at the end of the script, there is a special code block:

```
if __name__ == "__main__":
    # Call the main function if script has been run from the command line.
    main()
```

...which is not strictly necessary but helps to clarify that you expect this module to be run from the command line.

### PR DESCRIPTION
**Pull request by SoniaBarasa**
* Resolves #21 

This pull request will move the entire code into the main function. Previously, the main function code was empty.

### CODE REVIEW

File: gameofgame.py [hunk 1]
```diff
@@ -3,22 +3,15 @@
...
-    except (KeyError, ValueError):
-        continue
-
 
 # Function to choose a random country from the dictionary
+
+
 def choose_country():
     country = random.choice(list(countries_capitals.keys()))
     return country
```
Comment (ahernank) [inline gameofgame.py:12]:
```suggestion

```

File: gameofgame.py [hunk 2]
```diff
@@ -29,74 +22,93 @@ def check_guess(guess, country):
...
-    available_countries.remove(country)
+countries_capitals = {}
+for country in pycountry.countries:
+    try:
+        country_info = CountryInfo(country.alpha_2)
+        capital = country_info.capital()
+        if capital:
+            countries_capitals[country.name] = capital
+    except (KeyError, ValueError):
+        continue
```
Comment (ahernank) [inline gameofgame.py:36]:
```suggestion
def get_countries_capitals():
    countries_capitals = {}
    for country in pycountry.countries:
        try:
            country_info = CountryInfo(country.alpha_2)
            capital = country_info.capital()
            if capital:
                countries_capitals[country.name] = capital
        except (KeyError, ValueError):
            continue
    return countries_capitals
```

File: gameofgame.py [hunk 3]
```diff
@@ -29,74 +22,93 @@ def check_guess(guess, country):
...
+    player2_score = 0
+
+    # Defining the congratulatory message
+    congrats_message = "Congratulations Player {winner}, you are the winner!"
+
+    # defining string showing player's turn
+    player_turn_str = "Player {player_num}'s turn"
+
+    # Main game loop
+    # listing available countries
```
Comment (ahernank) [inline gameofgame.py:52]:
```suggestion
    # listing available countries
    countries_capitals = get_countries_capitals()
```

File: gameofgame.py [hunk 4]
```diff
@@ -29,74 +22,93 @@ def check_guess(guess, country):
...
+        # player 2 goes next
+        print(player_turn_str.format(player_num=2))
+        player2_guess = getpass.getpass(prompt=f"What is the capital of {country}? ")
+
+        # Reveal the correct answer
+        capital = countries_capitals[country]
+        print(f"The correct answer is {capital}.")
+
+        # Check the guesses and update the scores
+        if check_guess(player1_guess, country) and check_guess(player2_guess, country):
```
Comment (ahernank) [inline gameofgame.py:77]:
```suggestion
        if check_guess(player1_guess, country,  countries_capitals) and check_guess(player2_guess, country, countries_capitals):
```

File: gameofgame.py [hunk 5]
```diff
@@ -29,74 +22,93 @@ def check_guess(guess, country):
...
+            print("Both players guessed correctly!")
+            player1_score += 1
+            player2_score += 1
+        elif check_guess(player1_guess, country):
```
Comment (ahernank) [inline gameofgame.py:81]:
```suggestion
        elif check_guess(player1_guess, country, countries_capitals):
```

File: gameofgame.py [hunk 6]
```diff
@@ -29,74 +22,93 @@ def check_guess(guess, country):
...
+            print("Player 1 guessed correctly!")
+            print("Well done, you know your Geography well!")
+            player1_score += 1
+        elif check_guess(player2_guess, country):
```
Comment (ahernank) [inline gameofgame.py:85]:
```suggestion
        elif check_guess(player2_guess, country, countries_capitals):
```

**Review by ahernank — COMMENTED** [review]
Thanks, @SoniaBarasa. Suggesting some changes here. Note that for these to work, you will also need to update the function `check_guess(guess, country)` to include the countries+capitals dictionary as an argument (i.e. `check_guess(guess, country, countries_capitals`).

**Comment by SoniaBarasa** [comment]
Thank you very much folks for the comments. I am going to merge this pull request and close the issue now.

**Pull request merged**